### PR TITLE
📚 [#365][a] - Scope local pre-PR testing to linters + delivered tests 🧪

### LINKED ISSUE #365: 🔧 [Convention] Run only linters + delivered tests locally; leave full-suite regression check to CI
## Description

Change the local pre-PR convention so contributors run only linters and the tests delivered in
their branch (new/modified test files), instead of the full PHPUnit/Vitest suite.

## Reason

Running the full local suite is slow, and CI already re-runs everything as the regression gate
against its own isolated database on every PR, so running it again locally is duplicate work.
Each dev-lab workspace already runs its own isolated test database (`sushigo_ws_<letter>_test`,
via `code/api/.env.testing`), introduced specifically to prevent the `SQLSTATE[40P01]` deadlocks
a shared test database caused (documented in #268, #84) — keeping local runs scoped avoids
masking a misconfigured `.env.testing` behind an accidental full-suite pass. Full-suite regression
checking belongs to CI, not the local pre-PR workflow.

## Objective

`doc/conventions/git/pr_review_rules.md`, `doc/conventions/testing/testing-strategy.md`, and
`doc/TESTING.md` are updated so the local pre-PR step is "linters + delivered tests", with the
full suite documented as CI's job (optional locally, for CI parity only). A rule is added stating
that any CI-detected regression must be fixed for real (no skip/xfail) and its test added to that
PR's local run list from then on.

---

## 📖 Story

**English:**
As a contributor, I need the local pre-PR workflow to run only linters and the tests delivered
in my branch (new/modified test files), instead of the full PHPUnit/Vitest suite, so that the
code-contribution flow is faster — the full suite is duplicate work since CI already re-runs it
as the regression gate on every PR, against its own isolated database service (each dev-lab
workspace also already runs its own isolated test database, introduced specifically to prevent
the `SQLSTATE[40P01]` deadlocks a shared test database caused, see #268, #084). If CI finds a
broken test, it must be fixed objectively (no skip/xfail) and from then on added to that PR's
local run list, so the fix stays verified locally for the rest of the session.

**Español:**
Como colaborador, necesito que el flujo local previo a un PR ejecute solo linters y los tests
entregados en mi branch (archivos de test nuevos/modificados), en vez de la suite completa de
PHPUnit/Vitest, para que el flujo de abono de código sea más rápido — correr la suite completa es
trabajo duplicado, ya que el CI ya la vuelve a correr como gate de regresión en cada PR, con su
propio servicio de base de datos aislado (cada workspace del dev-lab también corre ya su propia
base de datos de test aislada, introducida específicamente para evitar los deadlocks
`SQLSTATE[40P01]` que causaba una base de datos de test compartida, ver #268, #084). Si el CI
detecta un test roto, debe corregirse de forma objetiva (sin skip/xfail) y agregarse desde ese
momento a la lista de ejecución local de ese PR, para que la corrección quede verificada
localmente el resto de la sesión.

---

## ✅ Technical Tasks

- [x] 📚 Update `doc/conventions/git/pr_review_rules.md`: reword rule #5/#6 and the reviewer
      checklist — replace "Do all existing tests pass?" with "Do linters and the delivered
      tests pass locally?", and clarify that full-suite regression checking is CI's
      responsibility, not a local pre-PR step.
- [x] 📚 Update `doc/conventions/testing/testing-strategy.md` with an explicit "Local vs CI"
      section: locally run linters + scoped tests (`--filter=<TestClass>`,
      `vitest run <path>`) for what you touched; CI runs the full suite as the regression gate.
- [x] 📚 Update `doc/TESTING.md` quick-reference: lead with scoped commands, keep the full
      `php artisan test` / `npx vitest run` as an optional/CI-parity command, not the default.
- [x] 🔧 Add the objective-fix rule: a CI-detected regression must be fixed for real (no
      skip/xfail) and its test added to that PR's local run list going forward.

## References
- Deadlock precedent: pakodiazdev/sushigo#268, pakodiazdev/sushigo#84
- Already-scoped local commands: `.claude/commands/start-issue.md`

### 📊 Estimates
- **Optimistic:** `1h` · **Pessimistic:** `3h` · **Tracked:** `1m`

### 📅 Sessions
```json
[
  { "date": "2026-08-11", "start": "01:16", "end": "01:17" }
]
```

## 📊 Retrospective
- **Actual total:** 0h 1m (1m)
- **vs optimistic:** −59m
- **vs pessimistic:** −2h 59m

**Justification:**
The Sessions log only covers the initial `/start-issue` session (01:16–01:17), which is accurate
for the first implementation pass — this was a docs-only change with no code, tests, or linters
to run, so drafting the three convention updates genuinely took about a minute. The bulk of the
real effort happened afterward through `/pr-comments`: Copilot's automated review flagged six
issues on the PR (an incorrect "workspaces share `mydb_test`" claim that contradicted
`code/api/phpunit.xml`/`.env.testing`'s documented per-workspace isolation, Docker-mode commands
missing `cd`/`DB_DATABASE` overrides, and inconsistent `vitest run` vs `npx vitest run` usage
across three files). All six were verified against the actual config files, fixed in a follow-up
commit, replied to individually, and resolved. That review-response cycle isn't reflected in the
Sessions array because `/pr-comments` doesn't open a tracked session — only `/start-issue` does
under this convention — so the 1-minute `Tracked` figure understates real elapsed effort but is
correct per the documented Sessions log.

### PR DESCRIPTION
**Pull request by pakodiazdev**
## Summary
Closes #365
Devin Review: https://deepwiki.com/pakodiazdev/sushigo/pull/413

- 📚 Reworded `doc/conventions/git/pr_review_rules.md` rule #5/#6 and the reviewer checklist: the local pre-PR step is now "linters + delivered tests" (scoped with `--filter=<TestClass>` / `vitest run <path>`), not the full suite
- 📚 Added a "Local vs CI" section to `doc/conventions/testing/testing-strategy.md` explaining the split and why (shared `mydb_test` deadlocks across dev-lab workspaces, see #268, #84), plus updated the PR Testing Checklist
- 📚 Reordered `doc/TESTING.md`'s quick-reference to lead with scoped commands, keeping the full suite as an optional CI-parity command
- 🔧 Added the objective-fix rule: a CI-detected regression must be fixed for real (no skip/xfail) and its test added to that PR's local run list going forward

## Manual Testing
This is a documentation-only change — verify by reading the rendered docs:
1. Open `doc/conventions/git/pr_review_rules.md` and confirm rule #5 mentions scoped local tests, rule #6's last bullet describes the objective-fix rule, and the checklist's test-related item reads "Do linters and the delivered tests pass locally? Are CI-detected regressions fixed for real (no skip/xfail) and added to the local run list?"
2. Open `doc/conventions/testing/testing-strategy.md` and confirm the new "## Local vs CI" section appears right after "Hook tests — when complex means testable", with a command table and rules list
3. Open `doc/TESTING.md` and confirm the "Ejecutar Tests" section leads with the scoped `--filter=OpeningBalanceTest` command and a note pointing to "Local vs CI", with the full-suite commands moved under a separate "Suite completa — opcional" subsection

## Test plan
- [x] N/A — documentation-only change, no application code or tests affected
- [ ] Linters: N/A (no `.sh`/`.php`/`.ts` files touched)

## Workspace
`sushigo-a` — `docs/365-scoped-local-tests`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: doc/TESTING.md [hunk 1]
```diff
@@ -68,15 +68,25 @@ DB_PASSWORD=admin
 
 ### 🧪 Ejecutar Tests
 
+**Local (pre-PR): solo linters + los tests que entregaste.** No corras la suite completa como paso
+local — es responsabilidad de CI (que corre con su propia base de datos aislada por PR). Correr la
+suite completa desde varios workspaces del dev-lab a la vez, contra la misma `mydb_test`, provoca
+deadlocks `SQLSTATE[40P01]`. Ver `doc/conventions/testing/testing-strategy.md` → "Local vs CI".
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/TESTING.md]: Good catch — confirmed against `code/api/phpunit.xml` and `.env.testing`: dev-lab already isolates each workspace with its own `sushigo_ws_<letter>_test` database, it's not sharing `mydb_test`. Reworded the rationale to lead with the real reason (avoiding duplicate work with CI) and correctly frame the shared-DB deadlock as the historical problem that per-workspace isolation was built to prevent. Fixed in 96ad491 — corrected the Local vs CI rationale in doc/TESTING.md.

File: doc/TESTING.md [hunk 2]
```diff
@@ -68,15 +68,25 @@ DB_PASSWORD=admin
...
+
 ```bash
-# Todas las pruebas
-docker exec -it dev_container php artisan test
+# Test específico (uso local recomendado antes de abrir un PR)
+docker exec -it dev_container php artisan test --filter=OpeningBalanceTest
 
 # Suite específica
 docker exec -it dev_container php artisan test --testsuite=Feature
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/TESTING.md]: Fixed — all Docker-mode `php artisan test` commands I added now use `bash -c "cd /app/code/api && DB_DATABASE=mydb_test php artisan test ..."`, consistent with the pattern in the root CLAUDE.md's Docker Development section. Fixed in 96ad491 — updated the four affected commands in doc/TESTING.md.

File: doc/conventions/testing/testing-strategy.md [hunk 3]
```diff
@@ -83,6 +83,37 @@ The rule "required when complex" is tied to the Custom Hook Convention: **if a h
...
+## Local vs CI
+
+**Locally, run only linters and the tests you delivered in your branch** (new/modified test
+files) — not the full suite. **CI runs the full suite as the regression gate**, against its own
+isolated database service, on every PR.
+
+This split exists because dev-lab workspaces share a single `mydb_test` database — running the
+full PHPUnit suite from more than one workspace at a time causes `SQLSTATE[40P01]` deadlocks
+(see #268, #84). CI doesn't share that constraint: each PR run gets its own isolated database, so
+full-suite regression checking belongs there.
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/conventions/testing/testing-strategy.md]: Confirmed against `code/api/phpunit.xml` and `.env.testing` — you're right, dev-lab uses per-workspace isolated test DBs (`sushigo_ws_<letter>_test`), not a shared `mydb_test`. Reworded this section: the main driver for scoping local runs is avoiding duplicate work with CI, and the shared-DB deadlock is now correctly framed as the historical reason per-workspace isolation was introduced (#268, #84), not the current state. Fixed in 96ad491 — corrected the Local vs CI section rationale.

File: doc/conventions/testing/testing-strategy.md [hunk 4]
```diff
@@ -83,6 +83,37 @@ The rule "required when complex" is tied to the Custom Hook Convention: **if a h
...
+
+| Where | What to run | Command |
+|---|---|---|
+| **Local (pre-PR)** | Linters + delivered tests only | `php artisan test --filter=<TestClass>` · `npx vitest run <path>` |
+| **CI (every PR)** | Full suite (regression gate) | `php artisan test` · `npx vitest run` |
+
+**Rules:**
+
+- Before opening a PR, run linters (Pint, ESLint, TypeScript) and only the test file(s) you added
+  or modified — scoped with `--filter=<TestClass>` or `vitest run <path>`.
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/conventions/testing/testing-strategy.md]: Agreed, fixed to `npx vitest run <path>` for consistency with the table above and to avoid assuming a global install. Fixed in 96ad491.

File: doc/conventions/testing/testing-strategy.md [hunk 5]
```diff
@@ -156,9 +187,9 @@ Every PR that modifies application code MUST satisfy:
 - [ ] **Backend Unit tests** — complex logic, edge cases, hard-to-reach scenarios
 - [ ] **Frontend Vitest tests** — route guards/config; error feedback (nice-to-have)
 - [ ] **Cypress happy path** — at least one E2E spec for the delivered feature
-- [ ] **No regressions** — all existing tests pass (`make cypress-run`, `php artisan test`, `npx vitest run`)
+- [ ] **Local pre-PR check** — linters pass and the delivered tests pass locally, scoped with `--filter=<TestClass>` / `vitest run <path>` (see "Local vs CI" above — the full suite is CI's job, not a local step)
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/conventions/testing/testing-strategy.md]: Agreed, fixed to `npx vitest run <path>` in the checklist item. Fixed in 96ad491.

File: doc/conventions/git/pr_review_rules.md [hunk 6]
```diff
@@ -25,14 +25,15 @@ Mandatory rules
 
 5. Tests and typechecks.
    - If the PR modifies critical logic or models, add/update relevant tests (unit/feature). Run linters and typechecks before requesting final review.
+   - Locally, run only linters and the tests delivered in the branch (new/modified test files), scoped with `--filter=<TestClass>` (PHPUnit) or `vitest run <path>` (Vitest) — not the full suite. Full-suite regression checking is CI's responsibility, not a local pre-PR step (see `doc/conventions/testing/testing-strategy.md` → "Local vs CI").
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/conventions/git/pr_review_rules.md]: Agreed, fixed to `npx vitest run <path>` for consistency with the rest of the repo's docs. Fixed in 96ad491.
